Hw2.2.accountant.work in hibernate

kpn1-ya2019 opened the pull request.

kpn1-ya2019 committed:
Привел к состоянию HW2.1 для DBC Template
Исправлены ошибки в тестах
Выполнен переход на JDBC Template

kpn1-ya2019 committed:
Перевёл на Hibernate Departments
Проверил тесты и поправил для getById, нужно править аналогично и остальные тесты

kpn1-ya2019 committed:
Поправил тесты для Department и проверил
Перевёл на Hibernate Employee

kpn1-ya2019 committed:
Перевёл на Hibernate PaymentType

kpn1-ya2019 committed:
Обавил аннотацию в репо других классов
Перевёл на Hibernate PaymentOrder

kpn1-ya2019 committed:
Перевёл на Hibernate AgrPaymentOrder

kpn1-ya2019 committed:
Устранил замечания PR для HW2.1

kpn1-ya2019 committed:
Исправлены наименования последовательностей в моделях
Модифицированы методы создания новых сущностей (generateNew...) - отказ от обработки ошибок NonUniqueResultException и NoResultException. Переход на использование Optional и обработку isEmpty.https://sysout.ru/ См. https://github.com/myluckagain/sysout/tree/master/spring-data-jpa-2

makkov reviewed (approved):
Павел, добрый день! Посмотрел, в целом смотрится верно
Единственное, в репозиториях я бы использовал не setter'ы, а добавил мапперы

kpn1-ya2019 committed:
Наведение порядка в бранче

kpn1-ya2019 committed:
Merge remote-tracking branch 'remotes/origin/master' into HW2.2.accountant.WorkInHibernate

# Conflicts:
#	.gitignore
#	.mvn/wrapper/maven-wrapper.properties
#	HWJavaBootCamp2024/.mvn/wrapper/maven-wrapper.properties
#	HWJavaBootCamp2024/mvnw
#	HWJavaBootCamp2024/mvnw.cmd
#	HWJavaBootCamp2024/pom.xml
#	HWJavaBootCamp2024/src/main/java/com/colvir/accountant/AccountantApplication.java
#	HWJavaBootCamp2024/src/main/java/com/colvir/accountant/model/Department.java
#	HWJavaBootCamp2024/src/main/java/com/colvir/accountant/model/Employee.java
#	HWJavaBootCamp2024/src/main/java/com/colvir/accountant/model/PaymentOrder.java
#	HWJavaBootCamp2024/src/main/java/com/colvir/accountant/repository/DepartmentRepository.java
#	HWJavaBootCamp2024/src/main/java/com/colvir/accountant/repository/EmployeeRepository.java
#	HWJavaBootCamp2024/src/main/java/com/colvir/accountant/repository/PaymentOrderRepository.java
#	HWJavaBootCamp2024/src/main/resources/application.properties
#	HWJavaBootCamp2024/src/test/java/com/colvir/accountant/AccountantApplicationTests.java
#	demo-spring-boot/.mvn/wrapper/maven-wrapper.properties
#	demo-spring-boot/mvnw
#	demo-spring-boot/mvnw.cmd
#	demo-spring-boot/src/test/java/com/colvir/demospringboot/DemoSpringBootApplicationTests.java
#	mvnw
#	mvnw.cmd
#	src/test/java/com/colvir/accountant/AccountantApplicationTests.java

kpn1-ya2019 committed:
уборал лишнее после мержа

kpn1-ya2019 committed:
Доработан PaymentTypeRepository. Вместо сеттера использую session.merge для слияния отсоединенной сущности
см. статью
https://www.baeldung.com/hibernate-save-persist-update-merge-saveorupdate
----
Как альтернатива добавлен маппер, котрый также может использоваться вместо сеттера

kpn1-ya2019 committed:
Доработан все остальные репозитории.
Вместо сеттера использую session.merge для слияния отсоединенной сущности см. статью
https://www.baeldung.com/hibernate-save-persist-update-merge-saveorupdate
+ target добавил в игнор
